src/main/java/com/thebois/models/world/structures/IBuildable.java

```diff
@@ -0,0 +1,50 @@
+package com.thebois.models.world.structures;
+
+import java.util.Collection;
+
+import com.thebois.models.inventory.items.IItem;
+
+/**
+ * IBuildable is an interface for interacting game entities that can be built.
+ */
+public interface IBuildable {
+
+    /**
+     * A collection of IItems that are needed for the completion of the IBuildable object.
+     *
+     * @return collection of needed items for construction.
+     */
+    Collection<IItem> neededItems();
+
+    /**
+     * Method for delivering an IItem to an IBuildable.
+     *
+     * @param deliveredItem the IItem to be delivered.
+     *
+     * @return if delivery was successful.
+     */
+    boolean deliverItem(IItem deliveredItem);
+
+    /**
+     * Returns a value between 0 and 1 based on how many of the needed IItems have been received.
+     *
+     * @return ratio of completion.
+     */
+    float builtStatus();
+
+    /**
+     * Returns true if builtStatus is equal to 1.
```

Comment:
This is describing implementation details. The javadoc should describe what the method represents, not how it does it. So it should say "Returns true if construction is completed."